src/elements/hx-tooltip/index.spec.js

```diff
@@ -0,0 +1,247 @@
+            const positionOffset = 20;
+            const offsetValue = component.POSITION_OFFSET;
+
+            expect(offsetValue).to.equal(positionOffset);
+        });
+
+        it('test for checking all positions', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const position = component.getAttribute('position');
+            const changePosition = 'right-middle';
+            component.setShadowPosition(changePosition);
+            const newPosition = component.shadowRoot.querySelector('div').getAttribute('position')
+
+            expect(newPosition).to.not.equal(position);
+            expect(newPosition).to.equal(changePosition);
+        });
+    });
+
+    describe('test $onConnect method', () => {
+        it('should have a randomly generated or assigned ID on render', async () => {
+            const fragment = /** @type {HXTooltipElement} */ await fixture(mockup);
+            const attr = fragment.hasAttribute('id');
+            const tooltipId = fragment.getAttribute('id');
+
+            expect(attr).to.be.true;
+            expect(tooltipId).to.not.be.null;
+        });
+
+        it('should have role attribute default to tooltip', async () => {
+            const roleAttr = 'tooltip';
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const attr = component.hasAttribute('role');
+            const role = component.getAttribute('role');
+
+            expect(attr).to.be.true;
+            expect(role).to.equal(roleAttr);
+        });
+    });
+
+    describe('test <hx-tooltip> getters and setters', () => {
+        it('should get *for* value', async () => {
+            const forValue = 'icon1';
+            const component = /** @type {HXTooltipElement} */ await fixture(mockup);
+            const attr = component.htmlFor;
+
+            expect(attr).to.equal(forValue);
+        });
+
+        it('should set *for* attribute value', async () => {
+            const component = /** @type {HXTooltipElement} */ await fixture(template);
+            const value = 'newIcon';
+            const attr = component.hasAttribute('for');
+
+            component.htmlFor = value;
+            const forValue = component.getAttribute('for');
+
+            expect(attr).to.be.false;
+            expect(forValue).to.equal(value);
+        });
+
+        it('should set shadow root to new position', async () => {
+            const fragment = /** @type {HXTooltipElement} */ await fixture(mockup);
+            const newPosition = 'bottom-center';
+
+            fragment.setShadowPosition(newPosition);
+
+            const tooltipPosition = fragment.shadowRoot.querySelector('div').getAttribute('position');
+
+            expect(tooltipPosition).to.equal(newPosition);
+        });
+
+        it('should get control element id of icon1', async () => {
```

Comment:
Please change test description based on the updates below.

Reply:
Changes have been done, Please review